lib/ast.js

```diff
@@ -91,9 +98,11 @@ function reference(base, prop) {
 }
 
 function cast(base, valueType) {
-  var result = nop(base);
+  var result = util.extend({}, base);
```

Comment:
I am not sure what is happening here. Value type can be undefined? Why we deleting it from the result. A comment would be helpful, I am unsure of the motivation for some of this code